### ISSUE TITLE
Add `severity` attribute to testinfra

### ISSUE THREAD

**Issue by blarghmatey** [mitodl/salt-extensions#6]
Should we add a new attribute to all testinfra parameters to set a severity flag which would then propagate to an associated event tag on failure?

For instance, it would read something like:

``` salt
test_some_file_stuff:
  testinfra.file:
    - name: foo.txt
    - exists:
        expected: True
        comparison: is_
        severity: low
```

If this failed it would fire to an event tag something like `salt/testinfra/failures/low/test_some_file_stuff`.

Thoughts @bdero?

**Comment by bdero** [comment]
Hmm that's an interesting idea, it seems like it would be pretty simple to do and it could be worth it once we get to the point where we have the scheduler taking care of stuff for us.

For the path, did you mean placing it under the salt root: `/salt/testinfra/failures/low/test_some_file_stuff` (note the leading slash)? Is this a good place for it, and do other plugins in the salt ecosystem do stuff like this?

**Comment by blarghmatey** [comment]
Yes, that's what I meant for the path. I was working from memory :)

As far as where to implement, yes I think it would probably make sense to do it in the testinfra module. I'm pretty sure there are some cases in which a state will trigger an event, such as the `onfail` option which is (I believe) universally available.

**Comment by blarghmatey** [comment]
For reference: https://docs.saltstack.com/en/latest/ref/states/requisites.html#onfail and https://docs.saltstack.com/en/latest/ref/states/requisites.html#onchanges

**Comment by bdero** [comment]
Cool. I was a bit ambiguous, but I partly meant to ask if it's a normal/sane thing for a plugin to write to files under the salt root.

**Comment by blarghmatey** [comment]
Ah, gotcha. Yeah, we would have to put some thought into the tag structure. I was just throwing that out as a rough example.
